lib/commands/general.js

```diff
@@ -186,22 +186,25 @@ commands.setUrl = async function (url) {
   return await iosCommands.general.setUrl.call(this, url);
 };
 
-commands.getDevicePixelRatio = iosCommands.device.getDevicePixelRatio;
 commands.getViewportRect = iosCommands.device.getViewportRect;
 
 commands.getScreenHeight = async function () {
   return (await this.getWindowSize()).height;
 };
 
+commands.getScreenInfo = _.memoize(async function () {
+  log.info("Getting screen info from WDA, and caching the result");
+  return await this.proxyCommand('/wda/screen', 'GET');
+});
+
 commands.getStatusBarHeight = async function () {
-  try {
-    let statusBar = await this.findNativeElementOrElements(`class name`, `XCUIElementTypeStatusBar`, false);
-    return (await this.getSize(statusBar)).height;
-  } catch (e) {
-    log.warn(`Could not detect status bar, assuming it is of zero height. ` +
-             `Original error: '${e.message}'`);
-    return 0;
-  }
+  const info = await this.getScreenInfo();
+  return info.statusBarSize.height;
+};
+
+commands.getDevicePixelRatio = async function () {
```

Comment:
this can be memoized though, since scale value is a constant

Reply:
ok, good idea. i thought status bar height would never change but you're right that apps have control over that.